### ISSUE TITLE
tidb-operator could not work with kubernetes 1.23 

### ISSUE THREAD

**Issue by STRRL** [pingcap/tidb-operator#4356]
## Bug Report

**What version of Kubernetes are you using?**

```
Client Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.1", GitCommit:"86ec240af8cbd1b60bcc4c03c20da9b98005b92e", GitTreeState:"clean", BuildDate:"2021-12-16T11:41:01Z", GoVersion:"go1.17.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.1", GitCommit:"86ec240af8cbd1b60bcc4c03c20da9b98005b92e", GitTreeState:"clean", BuildDate:"2021-12-16T11:34:54Z", GoVersion:"go1.17.5", Compiler:"gc", Platform:"linux/amd64"}
```

**What version of TiDB Operator are you using?**

v1.2.6

**What storage classes exist in the Kubernetes cluster and what are used for PD/TiKV pods?**

**What's the status of the TiDB cluster pods?**

```
k get pods
NAME                                       READY   STATUS             RESTARTS      AGE
tidb-controller-manager-84c5c7846c-lcmxp   1/1     Running            0             3m23s
tidb-scheduler-545dbcb74c-sk9jl            1/2     CrashLoopBackOff   4 (90s ago)   3m23s
```

```
 k describe pod tidb-scheduler-545dbcb74c-sk9jl
Name:         tidb-scheduler-545dbcb74c-sk9jl
Namespace:    tidb-admin
Priority:     0
Node:         hackathon-bigball/172.31.24.88
Start Time:   Fri, 07 Jan 2022 11:39:54 +0800
Labels:       app.kubernetes.io/component=scheduler
              app.kubernetes.io/instance=tidb-operator
              app.kubernetes.io/name=tidb-operator
              pod-template-hash=545dbcb74c
Annotations:  <none>
Status:       Running
IP:           10.0.2.62
IPs:
  IP:           10.0.2.62
Controlled By:  ReplicaSet/tidb-scheduler-545dbcb74c
Containers:
  tidb-scheduler:
    Container ID:  containerd://f1e2eb8bc3361851f422590637efbbafbaca31bf1f50efdf0b40d44393b32275
    Image:         pingcap/tidb-operator:v1.2.5
    Image ID:      docker.io/pingcap/tidb-operator@sha256:e9b87423721f4dd560a40c0d544b48f7105e75eb10470dfbd78862ee381bc584
    Port:          <none>
    Host Port:     <none>
    Command:
      /usr/local/bin/tidb-scheduler
      -v=2
      -port=10262
    State:          Running
      Started:      Fri, 07 Jan 2022 11:40:09 +0800
    Ready:          True
    Restart Count:  0
    Limits:
      cpu:     250m
      memory:  150Mi
    Requests:
      cpu:        80m
      memory:     50Mi
    Environment:  <none>
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-jkpd7 (ro)
  kube-scheduler:
    Container ID:  containerd://bff4e659aadb879438c224c1f0762c0cef6f97e49278fa44a45c80f568bfdb59
    Image:         k8s.gcr.io/kube-scheduler:v1.23.1
    Image ID:      k8s.gcr.io/kube-scheduler@sha256:8be4eb1593cf9ff2d91b44596633b7815a3753696031a1eb4273d1b39427fa8c
    Port:          <none>
    Host Port:     <none>
    Command:
      kube-scheduler
      --port=10261
      --leader-elect=true
      --lock-object-namespace=tidb-admin
      --lock-object-name=tidb-scheduler
      --scheduler-name=tidb-scheduler
      --policy-configmap=tidb-scheduler-policy
      --v=2
      --policy-configmap-namespace=tidb-admin
    State:          Waiting
      Reason:       CrashLoopBackOff
    Last State:     Terminated
      Reason:       Error
      Exit Code:    1
      Started:      Fri, 07 Jan 2022 11:43:20 +0800
      Finished:     Fri, 07 Jan 2022 11:43:20 +0800
    Ready:          False
    Restart Count:  5
    Limits:
      cpu:     250m
      memory:  150Mi
    Requests:
      cpu:        80m
      memory:     50Mi
    Environment:  <none>
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-jkpd7 (ro)
Conditions:
  Type              Status
  Initialized       True
  Ready             False
  ContainersReady   False
  PodScheduled      True
Volumes:
  kube-api-access-jkpd7:
    Type:                    Projected (a volume that contains injected data from multiple sources)
    TokenExpirationSeconds:  3607
    ConfigMapName:           kube-root-ca.crt
    ConfigMapOptional:       <nil>
    DownwardAPI:             true
QoS Class:                   Burstable
Node-Selectors:              <none>
Tolerations:                 node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                             node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:
  Type     Reason     Age                    From               Message
  ----     ------     ----                   ----               -------
  Normal   Scheduled  4m37s                  default-scheduler  Successfully assigned tidb-admin/tidb-scheduler-545dbcb74c-sk9jl to hackathon-bigball
  Normal   Pulling    4m36s                  kubelet            Pulling image "pingcap/tidb-operator:v1.2.5"
  Normal   Pulled     4m22s                  kubelet            Successfully pulled image "pingcap/tidb-operator:v1.2.5" in 13.947334894s
  Normal   Created    4m22s                  kubelet            Created container tidb-scheduler
  Normal   Started    4m22s                  kubelet            Started container tidb-scheduler
  Normal   Created    3m39s (x4 over 4m22s)  kubelet            Created container kube-scheduler
  Normal   Started    3m38s (x4 over 4m22s)  kubelet            Started container kube-scheduler
  Warning  BackOff    2m59s (x8 over 4m21s)  kubelet            Back-off restarting failed container
  Normal   Pulled     2m45s (x5 over 4m22s)  kubelet            Container image "k8s.gcr.io/kube-scheduler:v1.23.1" already present on machine
```

```
 k -n tidb-admin logs tidb-scheduler-545dbcb74c-sk9jl kube-scheduler
Usage:
  kube-scheduler [flags]

Misc flags:

      --config string            The path to the configuration file.
      --master string            The address of the Kubernetes API server (overrides any value in kubeconfig)
      --write-config-to string   If set, write the configuration values to this file and exit.

Secure serving flags:

      --bind-address ip                        The IP address on which to listen for the --secure-port port. The associated interface(s) must be reachable by the rest of the cluster, and by CLI/web clients. If blank or an unspecified address (0.0.0.0 or ::), all interfaces will be used. (default 0.0.0.0)
      --cert-dir string                        The directory where the TLS certs are located. If --tls-cert-file and --tls-private-key-file are provided, this flag will be ignored.
      --http2-max-streams-per-connection int   The limit that the server gives to clients for the maximum number of streams in an HTTP/2 connection. Zero means to use golang's default.
      --permit-address-sharing                 If true, SO_REUSEADDR will be used when binding the port. This allows binding to wildcard IPs like 0.0.0.0 and specific IPs in parallel, and it avoids waiting for the kernel to release sockets in TIME_WAIT state. [default=false]
      --permit-port-sharing                    If true, SO_REUSEPORT will be used when binding the port, which allows more than one instance to bind on the same address and port. [default=false]
      --secure-port int                        The port on which to serve HTTPS with authentication and authorization. If 0, don't serve HTTPS at all. (default 10259)
      --tls-cert-file string                   File containing the default x509 Certificate for HTTPS. (CA cert, if any, concatenated after server cert). If HTTPS serving is enabled, and --tls-cert-file and --tls-private-key-file are not provided, a self-signed certificate and key are generated for the public address and saved to the directory specified by --cert-dir.
      --tls-cipher-suites strings              Comma-separated list of cipher suites for the server. If omitted, the default Go cipher suites will be used.
                                               Preferred values: TLS_AES_128_GCM_SHA256, TLS_AES_256_GCM_SHA384, TLS_CHACHA20_POLY1305_SHA256, TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA, TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256, TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA, TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384, TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305, TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256, TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA, TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256, TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA, TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384, TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305, TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256, TLS_RSA_WITH_AES_128_CBC_SHA, TLS_RSA_WITH_AES_128_GCM_SHA256, TLS_RSA_WITH_AES_256_CBC_SHA, TLS_RSA_WITH_AES_256_GCM_SHA384.
                                               Insecure values: TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256, TLS_ECDHE_ECDSA_WITH_RC4_128_SHA, TLS_ECDHE_RSA_WITH_3DES_EDE_CBC_SHA, TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256, TLS_ECDHE_RSA_WITH_RC4_128_SHA, TLS_RSA_WITH_3DES_EDE_CBC_SHA, TLS_RSA_WITH_AES_128_CBC_SHA256, TLS_RSA_WITH_RC4_128_SHA.
      --tls-min-version string                 Minimum TLS version supported. Possible values: VersionTLS10, VersionTLS11, VersionTLS12, VersionTLS13
      --tls-private-key-file string            File containing the default x509 private key matching --tls-cert-file.
      --tls-sni-cert-key namedCertKey          A pair of x509 certificate and private key file paths, optionally suffixed with a list of domain patterns which are fully qualified domain names, possibly with prefixed wildcard segments. The domain patterns also allow IP addresses, but IPs should only be used if the apiserver has visibility to the IP address requested by a client. If no domain patterns are provided, the names of the certificate are extracted. Non-wildcard matches trump over wildcard matches, explicit domain patterns trump over extracted names. For multiple key/certificate pairs, use the --tls-sni-cert-key multiple times. Examples: "example.crt,example.key" or "foo.crt,foo.key:*.foo.com,foo.com". (default [])

Authentication flags:

      --authentication-kubeconfig string                  kubeconfig file pointing at the 'core' kubernetes server with enough rights to create tokenreviews.authentication.k8s.io. This is optional. If empty, all token requests are considered to be anonymous and no client CA is looked up in the cluster.
      --authentication-skip-lookup                        If false, the authentication-kubeconfig will be used to lookup missing authentication configuration from the cluster.
      --authentication-token-webhook-cache-ttl duration   The duration to cache responses from the webhook token authenticator. (default 10s)
      --authentication-tolerate-lookup-failure            If true, failures to look up missing authentication configuration from the cluster are not considered fatal. Note that this can result in authentication that treats all requests as anonymous. (default true)
      --client-ca-file string                             If set, any request presenting a client certificate signed by one of the authorities in the client-ca-file is authenticated with an identity corresponding to the CommonName of the client certificate.
      --requestheader-allowed-names strings               List of client certificate common names to allow to provide usernames in headers specified by --requestheader-username-headers. If empty, any client certificate validated by the authorities in --requestheader-client-ca-file is allowed.
      --requestheader-client-ca-file string               Root certificate bundle to use to verify client certificates on incoming requests before trusting usernames in headers specified by --requestheader-username-headers. WARNING: generally do not depend on authorization being already done for incoming requests.
      --requestheader-extra-headers-prefix strings        List of request header prefixes to inspect. X-Remote-Extra- is suggested. (default [x-remote-extra-])
      --requestheader-group-headers strings               List of request headers to inspect for groups. X-Remote-Group is suggested. (default [x-remote-group])
      --requestheader-username-headers strings            List of request headers to inspect for usernames. X-Remote-User is common. (default [x-remote-user])

Authorization flags:

      --authorization-always-allow-paths strings                A list of HTTP paths to skip during authorization, i.e. these are authorized without contacting the 'core' kubernetes server. (default [/healthz,/readyz,/livez])
      --authorization-kubeconfig string                         kubeconfig file pointing at the 'core' kubernetes server with enough rights to create subjectaccessreviews.authorization.k8s.io. This is optional. If empty, all requests not skipped by authorization are forbidden.
      --authorization-webhook-cache-authorized-ttl duration     The duration to cache 'authorized' responses from the webhook authorizer. (default 10s)
      --authorization-webhook-cache-unauthorized-ttl duration   The duration to cache 'unauthorized' responses from the webhook authorizer. (default 10s)

Deprecated flags:

      --contention-profiling           DEPRECATED: enable lock contention profiling, if profiling is enabled. This parameter is ignored if a config file is specified in --config. (default true)
      --kube-api-burst int32           DEPRECATED: burst to use while talking with kubernetes apiserver. This parameter is ignored if a config file is specified in --config. (default 100)
      --kube-api-content-type string   DEPRECATED: content type of requests sent to apiserver. This parameter is ignored if a config file is specified in --config. (default "application/vnd.kubernetes.protobuf")
      --kube-api-qps float32           DEPRECATED: QPS to use while talking with kubernetes apiserver. This parameter is ignored if a config file is specified in --config. (default 50)
      --kubeconfig string              DEPRECATED: path to kubeconfig file with authorization and master location information. This parameter is ignored if a config file is specified in --config.
      --lock-object-name string        DEPRECATED: define the name of the lock object. Will be removed in favor of leader-elect-resource-name. This parameter is ignored if a config file is specified in --config. (default "kube-scheduler")
      --lock-object-namespace string   DEPRECATED: define the namespace of the lock object. Will be removed in favor of leader-elect-resource-namespace. This parameter is ignored if a config file is specified in --config. (default "kube-system")
      --profiling                      DEPRECATED: enable profiling via web interface host:port/debug/pprof/. This parameter is ignored if a config file is specified in --config. (default true)

Leader election flags:

      --leader-elect                             Start a leader election client and gain leadership before executing the main loop. Enable this when running replicated components for high availability. (default true)
      --leader-elect-lease-duration duration     The duration that non-leader candidates will wait after observing a leadership renewal until attempting to acquire leadership of a led but unrenewed leader slot. This is effectively the maximum duration that a leader can be stopped before it is replaced by another candidate. This is only applicable if leader election is enabled. (default 15s)
      --leader-elect-renew-deadline duration     The interval between attempts by the acting master to renew a leadership slot before it stops leading. This must be less than or equal to the lease duration. This is only applicable if leader election is enabled. (default 10s)
      --leader-elect-resource-lock string        The type of resource object that is used for locking during leader election. Supported options are 'endpoints', 'configmaps', 'leases', 'endpointsleases' and 'configmapsleases'. (default "leases")
      --leader-elect-resource-name string        The name of resource object that is used for locking during leader election. (default "kube-scheduler")
      --leader-elect-resource-namespace string   The namespace of resource object that is used for locking during leader election. (default "kube-system")
      --leader-elect-retry-period duration       The duration the clients should wait between attempting acquisition and renewal of a leadership. This is only applicable if leader election is enabled. (default 2s)

Feature gate flags:

      --feature-gates mapStringBool   A set of key=value pairs that describe feature gates for alpha/experimental features. Options are:
                                      APIListChunking=true|false (BETA - default=true)
                                      APIPriorityAndFairness=true|false (BETA - default=true)
                                      APIResponseCompression=true|false (BETA - default=true)
                                      APIServerIdentity=true|false (ALPHA - default=false)
                                      APIServerTracing=true|false (ALPHA - default=false)
                                      AllAlpha=true|false (ALPHA - default=false)
                                      AllBeta=true|false (BETA - default=false)
                                      AnyVolumeDataSource=true|false (ALPHA - default=false)
                                      AppArmor=true|false (BETA - default=true)
                                      CPUManager=true|false (BETA - default=true)
                                      CPUManagerPolicyAlphaOptions=true|false (ALPHA - default=false)
                                      CPUManagerPolicyBetaOptions=true|false (BETA - default=true)
                                      CPUManagerPolicyOptions=true|false (BETA - default=true)
                                      CSIInlineVolume=true|false (BETA - default=true)
                                      CSIMigration=true|false (BETA - default=true)
                                      CSIMigrationAWS=true|false (BETA - default=true)
                                      CSIMigrationAzureDisk=true|false (BETA - default=true)
                                      CSIMigrationAzureFile=true|false (BETA - default=false)
                                      CSIMigrationGCE=true|false (BETA - default=true)
                                      CSIMigrationOpenStack=true|false (BETA - default=true)
                                      CSIMigrationPortworx=true|false (ALPHA - default=false)
                                      CSIMigrationvSphere=true|false (BETA - default=false)
                                      CSIStorageCapacity=true|false (BETA - default=true)
                                      CSIVolumeHealth=true|false (ALPHA - default=false)
                                      CSRDuration=true|false (BETA - default=true)
                                      ControllerManagerLeaderMigration=true|false (BETA - default=true)
                                      CustomCPUCFSQuotaPeriod=true|false (ALPHA - default=false)
                                      CustomResourceValidationExpressions=true|false (ALPHA - default=false)
                                      DaemonSetUpdateSurge=true|false (BETA - default=true)
                                      DefaultPodTopologySpread=true|false (BETA - default=true)
                                      DelegateFSGroupToCSIDriver=true|false (BETA - default=true)
                                      DevicePlugins=true|false (BETA - default=true)
                                      DisableAcceleratorUsageMetrics=true|false (BETA - default=true)
                                      DisableCloudProviders=true|false (ALPHA - default=false)
                                      DisableKubeletCloudCredentialProviders=true|false (ALPHA - default=false)
                                      DownwardAPIHugePages=true|false (BETA - default=true)
                                      EfficientWatchResumption=true|false (BETA - default=true)
                                      EndpointSliceTerminatingCondition=true|false (BETA - default=true)
                                      EphemeralContainers=true|false (BETA - default=true)
                                      ExpandCSIVolumes=true|false (BETA - default=true)
                                      ExpandInUsePersistentVolumes=true|false (BETA - default=true)
                                      ExpandPersistentVolumes=true|false (BETA - default=true)
                                      ExpandedDNSConfig=true|false (ALPHA - default=false)
                                      ExperimentalHostUserNamespaceDefaulting=true|false (BETA - default=false)
                                      GRPCContainerProbe=true|false (ALPHA - default=false)
                                      GracefulNodeShutdown=true|false (BETA - default=true)
                                      GracefulNodeShutdownBasedOnPodPriority=true|false (ALPHA - default=false)
                                      HPAContainerMetrics=true|false (ALPHA - default=false)
                                      HPAScaleToZero=true|false (ALPHA - default=false)
                                      HonorPVReclaimPolicy=true|false (ALPHA - default=false)
                                      IdentifyPodOS=true|false (ALPHA - default=false)
                                      InTreePluginAWSUnregister=true|false (ALPHA - default=false)
                                      InTreePluginAzureDiskUnregister=true|false (ALPHA - default=false)
                                      InTreePluginAzureFileUnregister=true|false (ALPHA - default=false)
                                      InTreePluginGCEUnregister=true|false (ALPHA - default=false)
                                      InTreePluginOpenStackUnregister=true|false (ALPHA - default=false)
                                      InTreePluginPortworxUnregister=true|false (ALPHA - default=false)
                                      InTreePluginRBDUnregister=true|false (ALPHA - default=false)
                                      InTreePluginvSphereUnregister=true|false (ALPHA - default=false)
                                      IndexedJob=true|false (BETA - default=true)
                                      JobMutableNodeSchedulingDirectives=true|false (BETA - default=true)
                                      JobReadyPods=true|false (ALPHA - default=false)
                                      JobTrackingWithFinalizers=true|false (BETA - default=true)
                                      KubeletCredentialProviders=true|false (ALPHA - default=false)
                                      KubeletInUserNamespace=true|false (ALPHA - default=false)
                                      KubeletPodResources=true|false (BETA - default=true)
                                      KubeletPodResourcesGetAllocatable=true|false (BETA - default=true)
                                      LocalStorageCapacityIsolation=true|false (BETA - default=true)
                                      LocalStorageCapacityIsolationFSQuotaMonitoring=true|false (ALPHA - default=false)
                                      LogarithmicScaleDown=true|false (BETA - default=true)
                                      MemoryManager=true|false (BETA - default=true)
                                      MemoryQoS=true|false (ALPHA - default=false)
                                      MixedProtocolLBService=true|false (ALPHA - default=false)
                                      NetworkPolicyEndPort=true|false (BETA - default=true)
                                      NodeSwap=true|false (ALPHA - default=false)
                                      NonPreemptingPriority=true|false (BETA - default=true)
                                      OpenAPIEnums=true|false (ALPHA - default=false)
                                      OpenAPIV3=true|false (ALPHA - default=false)
                                      PodAffinityNamespaceSelector=true|false (BETA - default=true)
                                      PodAndContainerStatsFromCRI=true|false (ALPHA - default=false)
                                      PodDeletionCost=true|false (BETA - default=true)
                                      PodOverhead=true|false (BETA - default=true)
                                      PodSecurity=true|false (BETA - default=true)
                                      PreferNominatedNode=true|false (BETA - default=true)
                                      ProbeTerminationGracePeriod=true|false (BETA - default=false)
                                      ProcMountType=true|false (ALPHA - default=false)
                                      ProxyTerminatingEndpoints=true|false (ALPHA - default=false)
                                      QOSReserved=true|false (ALPHA - default=false)
                                      ReadWriteOncePod=true|false (ALPHA - default=false)
                                      RecoverVolumeExpansionFailure=true|false (ALPHA - default=false)
                                      RemainingItemCount=true|false (BETA - default=true)
                                      RemoveSelfLink=true|false (BETA - default=true)
                                      RotateKubeletServerCertificate=true|false (BETA - default=true)
                                      SeccompDefault=true|false (ALPHA - default=false)
                                      ServerSideFieldValidation=true|false (ALPHA - default=false)
                                      ServiceInternalTrafficPolicy=true|false (BETA - default=true)
                                      ServiceLBNodePortControl=true|false (BETA - default=true)
                                      ServiceLoadBalancerClass=true|false (BETA - default=true)
                                      SizeMemoryBackedVolumes=true|false (BETA - default=true)
                                      StatefulSetAutoDeletePVC=true|false (ALPHA - default=false)
                                      StatefulSetMinReadySeconds=true|false (BETA - default=true)
                                      StorageVersionAPI=true|false (ALPHA - default=false)
                                      StorageVersionHash=true|false (BETA - default=true)
                                      SuspendJob=true|false (BETA - default=true)
                                      TopologyAwareHints=true|false (BETA - default=false)
                                      TopologyManager=true|false (BETA - default=true)
                                      VolumeCapacityPriority=true|false (ALPHA - default=false)
                                      WinDSR=true|false (ALPHA - default=false)
                                      WinOverlay=true|false (BETA - default=true)
                                      WindowsHostProcessContainers=true|false (BETA - default=true)
                                      csiMigrationRBD=true|false (ALPHA - default=false)

Metrics flags:

      --allow-metric-labels stringToString       The map from metric-label to value allow-list of this label. The key's format is <MetricName>,<LabelName>. The value's format is <allowed_value>,<allowed_value>...e.g. metric1,label1='v1,v2,v3', metric1,label2='v1,v2,v3' metric2,label1='v1,v2,v3'. (default [])
      --disabled-metrics strings                 This flag provides an escape hatch for misbehaving metrics. You must provide the fully qualified metric name in order to disable it. Disclaimer: disabling metrics is higher in precedence than showing hidden metrics.
      --show-hidden-metrics-for-version string   The previous version for which you want to show hidden metrics. Only the previous minor version is meaningful, other values will not be allowed. The format is <major>.<minor>, e.g.: '1.16'. The purpose of this format is make sure you have the opportunity to notice if the next release hides additional metrics, rather than being surprised when they are permanently removed in the release after that.

Logs flags:

      --experimental-logging-sanitization    [Experimental] When enabled prevents logging of fields tagged as sensitive (passwords, keys, tokens).
                                             Runtime log sanitization may introduce significant computation overhead and therefore should not be enabled in production.
      --log-flush-frequency duration         Maximum number of seconds between log flushes (default 5s)
      --log-json-info-buffer-size quantity   [Experimental] In JSON format with split output streams, the info messages can be buffered for a while to increase performance. The default value of zero bytes disables buffering. The size can be specified as number of bytes (512), multiples of 1000 (1K), multiples of 1024 (2Ki), or powers of those (3M, 4G, 5Mi, 6Gi).
      --log-json-split-stream                [Experimental] In JSON format, write error messages to stderr and info messages to stdout. The default is to write a single stream to stdout.
      --logging-format string                Sets the log format. Permitted formats: "json", "text".
                                             Non-default formats don't honor these flags: --add-dir-header, --alsologtostderr, --log-backtrace-at, --log-dir, --log-file, --log-file-max-size, --logtostderr, --one-output, --skip-headers, --skip-log-headers, --stderrthreshold, --vmodule.
                                             Non-default choices are currently alpha and subject to change without warning. (default "text")
  -v, --v Level                              number for the log level verbosity
      --vmodule pattern=N,...                comma-separated list of pattern=N settings for file-filtered logging (only works for text log format)

Global flags:

      --add-dir-header                   If true, adds the file directory to the header of the log messages (DEPRECATED: will be removed in a future release, see https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/2845-deprecate-klog-specific-flags-in-k8s-components)
      --alsologtostderr                  log to standard error as well as files (DEPRECATED: will be removed in a future release, see https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/2845-deprecate-klog-specific-flags-in-k8s-components)
  -h, --help                             help for kube-scheduler
      --log-backtrace-at traceLocation   when logging hits line file:N, emit a stack trace (default :0) (DEPRECATED: will be removed in a future release, see https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/2845-deprecate-klog-specific-flags-in-k8s-components)
      --log-dir string                   If non-empty, write log files in this directory (DEPRECATED: will be removed in a future release, see https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/2845-deprecate-klog-specific-flags-in-k8s-components)
      --log-file string                  If non-empty, use this log file (DEPRECATED: will be removed in a future release, see https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/2845-deprecate-klog-specific-flags-in-k8s-components)
      --log-file-max-size uint           Defines the maximum size a log file can grow to. Unit is megabytes. If the value is 0, the maximum file size is unlimited. (default 1800) (DEPRECATED: will be removed in a future release, see https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/2845-deprecate-klog-specific-flags-in-k8s-components)
      --logtostderr                      log to standard error instead of files (default true) (DEPRECATED: will be removed in a future release, see https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/2845-deprecate-klog-specific-flags-in-k8s-components)
      --one-output                       If true, only write logs to their native severity level (vs also writing to each lower severity level) (DEPRECATED: will be removed in a future release, see https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/2845-deprecate-klog-specific-flags-in-k8s-components)
      --skip-headers                     If true, avoid header prefixes in the log messages (DEPRECATED: will be removed in a future release, see https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/2845-deprecate-klog-specific-flags-in-k8s-components)
      --skip-log-headers                 If true, avoid headers when opening log files (DEPRECATED: will be removed in a future release, see https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/2845-deprecate-klog-specific-flags-in-k8s-components)
      --stderrthreshold severity         logs at or above this threshold go to stderr (default 2) (DEPRECATED: will be removed in a future release, see https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/2845-deprecate-klog-specific-flags-in-k8s-components)
      --version version[=true]           Print version information and quit

Error: unknown flag: --scheduler-name
```

**What did you do?**

Install tidb-operator following https://docs.pingcap.com/tidb-in-kubernetes/stable/get-started.

**What did you expect to see?**
tidb-opeator works well

**What did you see instead?**
container `kube-scheduler` in pod `tidb-scheduler-545dbcb74c-sk9jl` crashed, cause by a removed flag `--scheduler-name`

**Comment by DanielZhangQD** [comment]
@STRRL Please use the chart in the master branch for now, we will release v1.3.0-beta.1 next week that supports Kubernetes >=v1.22.
